Sources/Parser/Tokenizer.swift

```diff
@@ -52,6 +52,12 @@ public struct Tokenizer {
         } else {
           tokens.append(Token(kind: .literal(.decimal(.integer(num))), sourceLocation: sourceLocation))
         }
+      } else if component.hasPrefix("0x"){
+        // The token is an address literal.
+        let hex = component.dropFirst(2)
```

Comment:
I don't think we need to drop the leading "0x", except if you have a good reason

Reply:
Mainly related to the Int conversion - without it there is rightly no need